packages/atomic/src/components/commerce/atomic-product-numeric-field-value/atomic-product-numeric-field-value.ts

```diff
@@ -0,0 +1,119 @@
+import {Product} from '@coveo/headless/commerce';
+import {html, LitElement, nothing} from 'lit';
+import {customElement, property, state} from 'lit/decorators.js';
+import {when} from 'lit/directives/when.js';
+import {bindingGuard} from '@/src/decorators/binding-guard';
+import {bindings} from '@/src/decorators/bindings';
+import {createProductContextController} from '@/src/decorators/commerce/product-template-decorators.js';
+import {errorGuard} from '@/src/decorators/error-guard';
+import type {InitializableComponent} from '@/src/decorators/types';
+import {
+  defaultNumberFormatter,
+  NumberFormatter,
+} from '../../common/formats/format-common';
+import type {CommerceBindings} from '../atomic-commerce-interface/atomic-commerce-interface';
+import {parseValue} from '../product-template-component-utils/product-utils.js';
+
+/**
+ * @alpha
+ * The `atomic-product-numeric-field-value` component renders the value of a number product field.
+ *
+ * The number can be formatted by adding a `atomic-format-number`, `atomic-format-currency` or `atomic-format-unit` component into this component.
+ */
+@customElement('atomic-product-numeric-field-value')
+@bindings()
+export class AtomicProductNumericFieldValue
+  extends LitElement
+  implements InitializableComponent<CommerceBindings>
+{
+  /**
+   * The field that the component should use.
+   * The component will try to find this field in the `Product.additionalFields` object unless it finds it in the `Product` object first.
+   */
+  @property({type: String, reflect: true}) public field!: string;
+
+  @state() private product!: Product;
+  @state() private formatter: NumberFormatter = defaultNumberFormatter;
+  @state() private valueToDisplay: string | null = null;
+
+  private productController = createProductContextController(this);
+
+  @state() public bindings!: CommerceBindings;
+
+  @state() public error!: Error;
+
+  protected createRenderRoot() {
+    return this;
+  }
+
+  private handleNumberFormatEvent = (event: Event) => {
+    const customEvent = event as CustomEvent<NumberFormatter>;
+    customEvent.preventDefault();
+    customEvent.stopPropagation();
+    this.formatter = customEvent.detail;
+    this.updateValueToDisplay();
+  };
```

Comment:
Can we avoid the type casting by using CustomEvent straight in the listener/parameter 🤔